DXMainClient/Online/Connection.cs

```diff
@@ -324,6 +326,110 @@ private void HandleComm(object client)
+
+                        foreach (var serverIPAddress in serverIPAddresses)
+                        {
+                            var theIPAddress = serverIPAddress;
+
+                            if (availableServerAndLatencyDict.Any(item => item.Key.Host == theIPAddress.ToString()))
+                            {
+                                Logger.Log($"Skipped a duplicate IP from {serverName} ({theIPAddress}).");
+
+                                continue;
+                            }
+
+                            if (failedServerIPs.Contains(theIPAddress.ToString()))
+                            {
+                                Logger.Log($"Skipped a failed server {serverName} ({theIPAddress}).");
+
+                                continue;
+                            }
+
+                            var pingTask = new Task(() =>
+                            {
+                                Logger.Log($"Attempting to ping {serverName} ({theIPAddress}).");
+
+                                var pingReply = new Ping().Send(theIPAddress, MAXIMUM_LATENCY);
+
+                                if (pingReply.Status == IPStatus.Success)
+                                {
+                                    var pingInMs = pingReply.RoundtripTime;
+
+                                    Logger.Log($"The latency in milliseconds to the server {serverName} ({theIPAddress}): {pingInMs}.");
+
+                                    availableServerAndLatencyDict.Add(new Server(theIPAddress.ToString(), serverName, serverPorts), pingInMs);
+                                }
+                                else
+                                {
+                                    if (pingReply.Status == IPStatus.TimedOut)
+                                    {
+                                        Logger.Log($"Pinging the server {serverName} ({theIPAddress}) timed out!");
+                                    }
+                                }
+                            });
+
+                            pingTask.Start();
+                            pingTasks.Add(pingTask);
+                        }
+
+                        Task.WaitAll(pingTasks.ToArray());
+                    });
+
+                    dnsAndPingTask.Start();
+                    dnsAndPingTasks.Add(dnsAndPingTask);
+                }
+
+                Task.WaitAll(dnsAndPingTasks.ToArray());
+            }
+            catch (Exception ex)
+            {
+                if (ex is AggregateException)
+                {
+                    foreach (var innerEx in ((AggregateException)ex).Flatten().InnerExceptions)
+                    {
+                        Logger.Log("Unable to contact with the server. " + innerEx.Message);
+                    }
+                }
+                else
+                {
+                    Logger.Log("Unable to contact with the server. " + ex.Message);
+                }
```

Comment:
Ditto.

Reply:
Fixed in c448a0473f5abb25e469f6ee78691bace0932b62 .